### DISCUSSION TITLE
Water heater?

### DISCUSSION THREAD

**Discussion by DamijanD** [gekkekoe/esphome-ecodan-hp#79] [Announcements]
Not an expert but it looks to me that HomeAssistant has special WaterHeater module: 
`from homeassistant.components.water_heater import`
More can be checked in [custom_components/melcloud_custom/water_heater.py](https://github.com/ollo69/ha-melcloud-custom/blob/master/custom_components/melcloud_custom/water_heater.py)

Would it make sense to implement this?

**Comment by gekkekoe** [comment]
@F1p actually looked at this yesterday, it does not really has any compelling features.
Also this project is an esphome project and the components are (automatically) translated to HA. Esphome does not have a water heater component. I modeled it as a climate, but when you look at the water heater, it's almost the same.

**Reply by DamijanD** [reply]
A OK. Like I said I am no expert and that python code reminded me of ha components.

**Reply by F1p** [reply]
It would make more sense if immersion was controllable, but as it isn't it's not particularly compelling

**Reply by DamijanD** [reply]
Can you explain why it would be more compelling if we could control immersion heater? (I don't even have it 😄)

With melcloud integration I used `Water heater` and set it to `force_hot_water` in automation when some conditions were met (solar power,...). Now with esphome I will use `Switch 'Turn on' on Ecodan heatpump Force DHW` (already set but too bad weather to meet conditions to really test it in action).

**Reply by gekkekoe** [reply]
Force dhw ignores eco setting.

you can work around this by triggering the via tempdrop (temp increase die setpoint)

Also heating via heatpump is way more efficient. I also don't have an immersion, but some ppl use it with solar or avoid pause (when doing doing dhw it cannot heat)

**Reply by DamijanD** [reply]
A I didn't know that force dhw ignores eco setting. Interesting (didn't see that in manuals...)
I will just keep on using force DHW command as I have DHW on scheduler (prohibited in the most expensive hours) and this tempdrop "hack" wouldn't work... I will keep an eye on force and normal dhw heating to see what is the actual difference in power consumption. 
At glance it looks like there is 500W difference (only heating marked with an arrow is force - all others are in eco mode):
![image](https://github.com/user-attachments/assets/ff738cb6-3ed0-479b-95bf-98f41a392e45)
